src/pyxplor/plot_numeric.py

```diff
@@ -49,13 +52,68 @@ def plot_numeric(
 
     Returns
     -------
-    None
-
+    tuple
+        A tuple containing the `fig` and `ax` objects.
 
     Examples
     --------
     numeric = ["numeric_var1", "numeric_var2"]
-    plot_numeric(df, numeric, "hist+kde", 30, 10, (15, 20))
+    plot_numeric(df, numeric, "hist+kde", 30, 10, (10, 15))
     """
+    # Get numeric columns from the dataframe
+    numeric_columns = input_df.select_dtypes(include=['number']).columns.tolist()
+
+    # Filter numeric columns based on the provided list_of_variables
+    numeric_columns = [var for var in list_of_variables if var in numeric_columns]
+
+# Check if there are any numeric columns to plot
+if not numeric_columns:
+    print("No valid numeric columns found in the provided list_of_variables.")
+    return
+
+# Calculate the number of rows and columns dynamically
+total_plots = len(numeric_columns)
+rows = math.ceil(math.sqrt(total_plots))
+cols = math.ceil(total_plots / rows)
+
+# Create subplots in a grid
+fig, ax = plt.subplots(rows, cols, figsize=figsize)
+ax = ax.flatten()
```

Comment:
This line throws an error if ax only has one element. I added the following to mitigate this!

```suggestion
    if len(list_of_variables) > 1:
        ax = ax.flatten()
```